packages/design-system-react/src/components/text/Text.types.ts

```diff
@@ -0,0 +1,102 @@
+export type TextProps = {
+  variant?: TextVariant;
+  children: React.ReactNode;
+  className?: string;
+  fontWeight?: FontWeight;
+  fontStyle?: FontStyle;
+  textTransform?: TextTransform;
+  textAlign?: TextAlign;
+  overflowWrap?: OverflowWrap;
+  ellipsis?: boolean;
+  as?: ValidTag;
+  color?: TextColor;
+};
+export enum TextVariant {
+  // Display Sizes
+  DisplayLg = 'display-lg',
+  DisplayMd = 'display-md',
+
+  // Heading Sizes
+  HeadingLg = 'heading-lg',
+  HeadingMd = 'heading-md',
+  HeadingSm = 'heading-sm',
+
+  // Font Sizes
+  BodyLg = 'body-lg',
+  BodyMd = 'body-md',
+  BodySm = 'body-sm',
+  BodyXs = 'body-xs',
+}
+
+export enum TextColor {
+  TextDefault = 'text-default',
+  TextAlternative = 'text-alternative',
+  TextMuted = 'text-muted',
+  OverlayInverse = 'text-overlay-inverse',
+  PrimaryDefault = 'text-primary-default',
+  PrimaryInverse = 'text-primary-inverse',
+  ErrorDefault = 'text-error-default',
+  ErrorAlternative = 'text-error-alternative',
+  ErrorInverse = 'text-error-inverse',
+  SuccessDefault = 'text-success-default',
+  SuccessInverse = 'text-success-inverse',
+  WarningDefault = 'text-warning-default',
+  WarningInverse = 'text-warning-inverse',
+  InfoDefault = 'text-info-default',
+  InfoInverse = 'text-info-inverse',
+  Inherit = 'text-inherit',
+  Transparent = 'text-transparent',
```

Comment:
Adding some descriptions with some best color usage would be helpful

Reply:
Good suggestion! Updated https://github.com/MetaMask/metamask-design-system/pull/110/commits/29ae62037dc1454a7d9b4186a3dc51fbc30944b8